src/controller.js

```diff
@@ -0,0 +1,149 @@
+(function exportController() {
+    class Controller {
+        constructor(ship) {
+            this.ship = ship;
+
+            this.initialiseSea();
+
+            if (ship.itinerary.ports.length > 0) {
+                this.hud();
+            }
+
+            document.querySelector('#sailbutton').addEventListener('click', () => {
+                this.setSail();
+            });
+        }
+
+        initialiseSea() {
+            const backgrounds = [
+                './image/water0.png',
+                './image/water1.png'
+            ];
+
+            let backgroundIndex = 0;
+            window.setInterval(() => {
+                document.querySelector('#viewport').style.backgroundImage = `url('${backgrounds[backgroundIndex % backgrounds.length]}')`;
```

Comment:
Long line like this can be reformatted as they dont shoot off the end of the screen:
```
                document.querySelector(
                    "#viewport"
                ).style.backgroundImage = `url('${
                    backgrounds[backgroundIndex % backgrounds.length]
                }')`;
```